### ISSUE TITLE
Multiple boards per repository

### ISSUE THREAD

**Issue by paavonelimarkka** [ZenHubIO/support#139]
Could it be possible to have multiple boards per repository (or is there already)? I would like to have a few boards for my repository for different purposes. For example handling business goals, epics and user stories in different board than simple tasks.

**Comment by morewry** [comment]
I tend to use milestones for this and have mostly found it adequate. The problem being an issue can only be in one milestone at a time.

**Comment by erichexter** [comment]
@aupright  team still working on this?

**Comment by aupright** [comment]
Hey @erichexter this is definitely still on our radar moving forward! I have to apologize for my comment above - I misunderstood your original request to be similar to #85 (multi-repo boards vs. multi-board repos). Having said that, this is still a feature that we would like to offer and will be exploring further once we have shipped multi-repo boards!

Apologies again for any confusion on my part!

**Comment by erichexter** [comment]
ok, thanks for the clarification.

**Comment by gustvao** [comment]
+1

**Comment by snario** [comment]
+1

**Comment by vgupta2** [comment]
+1  Without this zenhub isn't that helpful for me

**Comment by erichexter** [comment]
To work around this, I have ended up writing webhooks to move issues
between to repositories. Sucks, but I have made work

Eric Hexter

blog | http://Hex.LosTechies.com
info | http://www.linkedin.com/in/erichexter

**Comment by reubenpressman** [comment]
+1

**Comment by hadrienl** [comment]
It's a must have, just to be able to manage issues and PR with a different process.

**Comment by jandre** [comment]
+1 I really want to use this vs Trello, and the ability to have multiple boards ([see this workflow](https://teamgaslight.com/blog/beyond-pivotal-tracker-managing-software-projects-with-trello)) is really hampering my ability to do proper product management

**Comment by otint** [comment]
+1

**Comment by rmacklin** [comment]
I'm _very_ intrigued by ZenHub. Looking for a new tool, I knew I wanted something based around GitHub issues to eliminate duplicate entry/multiple "sources of truth". One simple search revealed that there's a lot out there (huboard, waffle, codetree, blossom, sweepboard, and zenhub are just the first few). I find ZenHub's approach of injecting additional features directly into GitHub's UI really promising for centralizing our workflow. (You guys are obviously well aware of all the benefits this brings.)

That said, not being able to have multiple boards per repository (or, really, per group-of-linked-repositories) hugely reduces the value my organization can get from ZenHub.

@ppat explained a common scenario in https://github.com/ZenHubIO/support/issues/427#issuecomment-153123359:
> Repo1 and Repo2 are part of a single project so you need to see them on the same board. But Repo2 contains a codebase that is shared and used by many other projects.

This is exactly the situation we deal with. Pretty much all of our teams contribute to one big repository, and many of them also contribute to at least one additional smaller repository. When it comes to workflow, our teams act independently (i.e. each team agrees how they want to work, and it can be completely different from how other teams operate). To support this, each team _needs_ to be able to define their own pipeline columns.

With ZenHub's existing feature set, we'd have to link together the union of all repositories worked on by any team and create the union of all pipeline columns used by any team. With enough filtering capabilities, that _could_ actually work. But the problem is: you can't specify different filter constraints per repository. I'd need to be able to filter by just my team's issues in our big shared repository while not filtering anything out of my team's small repository. This isn't currently possible. Even if it were, having the union of all teams' pipeline columns is still bad because you can only collapse columns your team doesn't use: you can't hide them entirely (as far as I know). It also just doesn't make much sense to link the union of all repos together when in reality no one would be looking at more than a handful of repos at a time.

Allowing "team-based" boards would solve the issue perfectly. You'd create a board for "Team A". You'd link only the repos that Team A works with. You'd add only the pipeline columns that Team A uses. Each repo could be part of the linked-repo-group of any board. When you navigate to "Boards" from a repo, you'd select Team A from the list of boards that repo belongs to (and your selection would of course be remembered the next time you visit). Now you'd have all the flexibility in the world. And it'd be enabled just by the addition of that one extra concept of a "team". (What I'm calling a "team" is just a group of people. Any group of people--it can cross any boundaries created by what your organization may already designate as "teams".)

**Comment by meggiebot** [comment]
This would be helpful for us since multiple teams may use the same repo in different ways. Each team would like to be able to customize their board for what they are interested in. Currently I have 3 teams sharing 1 board with 4 repos.

**Comment by azenMatt** [comment]
@rmacklin @meggiebot and others, 

In these situations, would you still want the issue to exist 'only once' or are you looking to create duplicate issues - unique to each board? 

To give an example. 

Team A create a board from Repo 'API', creating pipelines 1,2,3

Team B create a board from Repo 'API', creating pipelines 4,5,6


If team A move 'API issue #1' from pipeline 1 -> 2 , does this get reflected on Team B's board? 

If team B close 'API issue #1', does this also close it for Team A? 


In principle, I am all for the idea of being able to create more boards. But given that GitHub issues have a strict mapping to one repo, I feel it could cause a lot of confusion if boards no longer obey the same mapping. 

--------------------

Would you say the teams are looking to leverage issues for what they enable teams to share - comments, assignee information, labels etc - but do not want to share work flow information? 

or does being able to share comments etc not matter either?

**Comment by meggiebot** [comment]
@azenMatt 

Do you mean actually duplicating issues in the repo? Or just across boards? I would not want to duplicate the issues in the repo, just have them show up on multiple boards.

In the case where the pipelines do not match, I would not change the issue. But if the pipelines match, I would change the issues.

In the case where an issue is closed, I would close it on both boards.

From our perspective a board is just a view into a repo, like a dashboard. It makes sense to me to be able to view the data different ways depending on which team you are in.

If it would require duplicating issues (new comments, assignees, etc) this would definitely not work for us.

-------------

Our use case is that we have one major repo that multiple teams use. Issues often move between teams as they are being worked on (which we manage with functional area and team labels). However, not every team wants to see the same thing on their board.

We are currently getting around this by the teams bookmarking a board view based on filters that they are interested in (team label, milestone, severity label, etc).  But not ever team has the same workflow; so they don't all want the same pipelines. Currently this is a problem since a repo has one board and one set of pipelines.

The other way we are getting around this is by creating team-specific repos with their own board. But this is also a problem because they cannot merge the main repo with their team repo, so they use multiple boards to track all of their issues.

-------------

I can see how this starts to become a problem where you have an issue in a repo showing up on multiple boards with different pipelines. I am not sure the solution, I just know my teams would like to be able to customize their "dashboard."

**Comment by azenMatt** [comment]
got it - thanks for the feedback :smile:

**Comment by rmacklin** [comment]
@azenMatt 

For the most part, our teams work independently. So, most of those questions you listed wouldn't need to be solved for us to use multiple boards. Team A would be the only team working on API issue #1 -- even if they enlist help from a member of another team, the issue would still be "owned" by Team A. If there is a large project that involves multiple teams, it would be broken down into smaller pieces that the individual teams can work on separately, and the big "multi-team" issue wouldn't just get closed after all the individual pieces have been closed.

That said, assuming we did need an issue to appear on more than one team's board:
> In these situations, would you still want the issue to exist 'only once' or are you looking to create duplicate issues - unique to each board?

The issues would exist only once. Definitely wouldn't want duplicate issues, nor do I really see why that would be necessary (more on that at the end of this comment).

> If team A move 'API issue #1' from pipeline 1 -> 2 , does this get reflected on Team B's board?

I would say that each board should manage the pipeline state of the issues it displays. That is, an issue can be in one pipeline column per board, and these are independent of each other. Except for the special closed column:

> If team B close 'API issue #1', does this also close it for Team A?

Yes.

> In principle, I am all for the idea of being able to create more boards. But given that GitHub issues have a strict mapping to one repo, I feel it could cause a lot of confusion if boards no longer obey the same mapping.

I don't really follow here. I definitely am not suggesting that issues belong to more than one repo. We should definitely stick with GitHub's paradigm of issues being associated with one repo. My proposal was to add an extra layer between boards and repos (the "team" layer): a board is associated with a "team", and this board can show an aggregate view of the issues from multiple linked repos (independently filterable).

**Comment by meggiebot** [comment]
> I would say that each board should manage the pipeline state of the issues it displays. That is, an issue can be in one pipeline column per board, and these are independent of each other. 

I like this idea. Maybe each issue can only have one pipeline associated with it and only appears on boards with that pipeline. 

On boards where issues don't appear due to the pipeline not being on the board, there could be some kind of note about "5 issues do not appear on this board due to the pipeline configuration" that you could click for a list.

**Comment by rmacklin** [comment]
@meggiebot I don't think that what you're proposing is what I suggested, though I think that could work too.

What I had meant was, _if you need an issue to display on multiple boards_ (and again, this isn't really a use case for us because our teams work on independent stories), then Team A's Board could have that issue in any of Team A's pipeline columns (other than closed) and Team B's Board could have the same issue in any of Team B's pipeline columns (other than closed). Right now, in ZenHub an issue can only be in one pipeline column. With multiple boards, issues could be in one pipeline column per board in ZenHub. Maybe I'm not making any sense (sorry!)

**Comment by meggiebot** [comment]
@rmacklin I think I get what you are saying, though I don't know if it would work. In that case an issue that showed up on two boards would have 2+ pipelines. I don't know how that would be displayed on the issue detail view.

I actually think the idea you inspired me to have ( :wink: ) would work. Curious to see what the ZH team thinks--and I think it would work both of our use cases (where you do and don't need issues to display on multiple boards).

**Comment by rmacklin** [comment]
> I don't know how that would be displayed on the issue detail view.

Fair point. I actually forgot it's displayed on the issue view, as I look at the board to see where things are. I think it could still work, by showing the state it was in on the last board you looked at (with a note that it is in different state on another board). Again, though, for my organization this wouldn't actually be a problem because issues would only really move through one team's board. So you're right that your suggestion would also suit me fine.

**Comment by apatrida** [comment]
Would this also be solved as "multiple swim lanes per repo" which cover some of the use cases

**Comment by azenMatt** [comment]
> Would this also be solved as "multiple swim lanes per repo" which cover some of the use cases

Could help for some use-cases, but the vast majority of teams I speak to about this the problem stems from multiple teams using the same repo - but each team wanting unique pipelines.

**Comment by davidguthu** [comment]
+1 for multiple board in one repo.

**Comment by kevinSuttle** [comment]
I'm not sure if this is the same: but a milestone spanning multiple repos would be ideal too. I'll happily move this to another issue if so.

**Comment by azenMatt** [comment]
Hi @kevinSuttle - We don't have an issue exactly for that case, but I would group it with #298 Sync labels in merged boards. 

As per my comment on that issue here - https://github.com/ZenHubIO/support/issues/298#issuecomment-185507248 this is something that we have on our roadmap for the next quarter

**Comment by kevinSuttle** [comment]
:+1:

**Comment by georgebonnr** [comment]
> This would be helpful for us since multiple teams may use the same repo in different ways. Each team would like to be able to customize their board for what they are interested in. Currently I have 3 teams sharing 1 board with 4 repos.

:+1:  for our org as well

**Comment by simundi** [comment]
+1

**Comment by bergjs** [comment]
+1

**Comment by wezell** [comment]
+1 on this, as I am not sure how else to use zenhub for multiple teams working on a single piece of (large) software.  At any given time, the teams can be working on the same release, or on separate releases.  Each team and PM need a board to track all that.  



Were for this feature, I think we'd jump aboard

**Comment by azenMatt** [comment]
a question on this feature, 

Would the ability to define a 'team' or 'squad' help here? 

Making it easy to quickly filter a board by just by the issues relevant to that team.

**Comment by kevinSuttle** [comment]
The GitHub Org/Team feature could be leveraged if I understand correctly. 
That way, ZenHub can just tap into the APIs.

**Comment by bergjs** [comment]
@azenMatt I would say not really. I would need a separated board so I can structure it differently. For example in Trello we use a board only for collecting ideas and the design and specification process. A ticket that is "done" there will go to "to do" in the development board. I don't think this could be realized with filtering.

**Comment by erichexter** [comment]
This is my use case as well

On Friday, March 11, 2016, Jonas Bergert <notifications@github.com> wrote:

> @azenMatt <https://github.com/azenMatt> I would say not really. I would
> need a separated board so I can structure it differently. For example in
> Trello we use a board only for collecting ideas and the design and
> specification process. A ticket that is "done" there will go to "to do" in
> the development board. I don't think this could be realized with filtering.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/ZenHubIO/support/issues/139#issuecomment-195285228>.
>


-- 
Eric Hexter

blog | http://Hex.LosTechies.com
info | http://www.linkedin.com/in/erichexter

**Comment by meggiebot** [comment]
@azenMatt our team would definitely use the "team" or "squad" feature. We currently handle this through labels. Each team has a label and we sort the board by the label for that team. 

Optimally if we could put users into groups we could filter that way, since the engineers often forget the team label :)

**Comment by thaJeztah** [comment]
Having multiple boards would be a big improvements. We would like to use different "kind" of boards, using different pipelines. Filtering doesn't help there, because the boards should have a different layout. The current tab is called "Boards" so I was looking for a button to Switch between boards, or creating a new one, only to discover that's not a feature (yet).

**Comment by gregzuro** [comment]
+1

**Comment by ebordeau** [comment]
+1 

We have 2 repos for tracking issues... one for "stories", which we'd love to convert to epics, and one for tasks/defects. They have very different pipelines, so we didn't merge their boards. 

Our pipeline for epics is more for management to track where a particular epic is deployed - Upcoming (epics we're considering for the next milestone), Development, Integration (staging/QA), Production:
![image](https://cloud.githubusercontent.com/assets/4596191/14984858/daaa7162-1112-11e6-81b6-2e05c86b84fb.png)


Our dev issues board (tasks/defects) uses the default pipeline:
![image](https://cloud.githubusercontent.com/assets/4596191/14984865/e0cef734-1112-11e6-97d5-84fe14294ef8.png)

If we could have 2 boards, one for epics and one for tasks/defects, we could keep our issues in one repo and take advantage of the new epics.

**Comment by mneuss** [comment]
+1

**Comment by wrobbinz** [comment]
👍

**Comment by micheldoumit** [comment]
+1

**Comment by bergjs** [comment]
Wow, thanks for the info @gustvao. I didn't realize Github has boards now. Doesn't this make Zenhub obsolete now already?

**Comment by shirey** [comment]
+1

**Comment by markballinger** [comment]
As per some of the comments above, I want to create a product roadmap, which will span multiple sprint cycles. For example, putting Epics on the Roadmap with pipelines for each quarter, then using a board per sprint to manage the work specific to each one.

**Comment by erichexter** [comment]
Seems like the best answer for this is using GitHub projects . This request
has been out there for so long, it's clearly not on the roadmap

**Comment by aamorozov** [comment]
+1

**Comment by gladen** [comment]
@azenMatt are we gonna get multiple boards per repository? If yes, when (approximately)?

**Comment by stwirth** [comment]
+1

**Comment by azenMatt** [comment]
> are we gonna get multiple boards per repository? If yes, when (approximately)?

Hi @gladen - short answer - yes, Q3ish (the sooner the better!) 

The longer answer.....

This is an incredibly important piece of feedback to us, not only is this thread one of the biggest on this repo, but in conversations with (existing and potential) customers it comes up repeatedly. So to me, it is clearly something that we need to action. 

In terms of time frame, while it is high on the 'impact' scale, it is also high on the 'complexity' scale! Right now, the team is focused on #156 which is on track for the near future. Once this is done, reworking ZenHub's 'merge repository' workflow to both 1) make it much more user friendly 2) include the concept of multiple boards per repository become the highest priority 'big' feature. 

I'l add a note back here when we get started on this so everyone can get a better idea of the time frame

**Comment by trosman** [comment]
I need multiple boards for our project. I would say is the most important missing feature of Zenhub:)

**Comment by andrehagemeier** [comment]
@azenMatt is Q3ish still realistic? We're looking at how we manage a development team for ~250 developers and Boards work fine, but we need different levels of Boards (detailed level team views <-> high level release views) spanning various numbers of repositories (and even github orgs). 
Q3 would be fine, otherwise we'd have to solve this problem differently. Would appreciate an update

**Comment by neilsmind** [comment]
@azenMatt Given that a cross browser and mobile web app has been released for #156, have you started multi-board repos? Would love to see this functionality.

**Comment by lauracpierre** [comment]
@azenMatt Trying to see if you've made any progress on this? This would allow us to expand our team on Zenhub. 
Thanks and keep up the good work!

**Comment by azenMatt** [comment]
Hi @lauracpierre @neilsmind 

This remains one of most requested features for ZenHub via both this issue, emails, and conversations with customers. 

We are working through a couple of different design ideas on how to deliver this and will make sure to get feedback from this group and to keep you all updated before we move anything forward. I don't have an idea of timeline just yet - but once I do I will add an update here

**Comment by WebDevJL** [comment]
+1

**Comment by DanielePucci** [comment]
👍

**Comment by nunoguedelha** [comment]
Hi @azenMatt, I would like to add a very simple example where the **"multi-board repos"** feature is essential in a workflow involving repos distributed into multiple boards:
If I want to specify the dependency between two issues, I can only do it if they belong to repos included in the same board. So, by that time, if two issues ended up being tracked in different boards, there's no simple way I can add that dependency, without disconnecting one of them and thus breaking the workflow.

**Comment by knlyons** [comment]
Would it be possible to merge the board, without breaking an existing merged repo connection?

My team works cross-org and has repos in each respective org. We'd like a super board to pull out these repos, but we can't merge the repos as of now because they are already merged in their current orgs. Basically, we need an overall view across these orgs.

**Comment by kendramoroz** [comment]
Hey @knlyons thanks for touching base on Board merges - currently a Board can be merged with/within 1 workspace. This means if the repo is already connected within a multi-repo Board, if connected with a seperate repo, it will be disconnected from the other workspace. You can bring in the other repo into the already connected workspace, but the pipelines would have to match across all repos so this might not be ideal if it's two seperate teams here. 

We're currently working on a solution to have multiple Boards per repo, so we'll be sure to update this Issue once we have more to share. 

Let me know if you have any questions around best practices for your use case via kendra@zenhub.com and we can setup some time to dig in more prior to the launch of updates to the Boards 🚀

**Comment by norazulick** [comment]
+100 (commenting on behalf of everyone in my company using zenhub enterprise!)

**Comment by KingJammin** [comment]
+1

**Comment by valentinweber** [comment]
+1

**Comment by MilesRomney** [comment]
+1

**Comment by azenMatt** [comment]
Hi Everyone, 

We are working on this change at the moment and would love some feedback to make sure that we are going to hit the core use-cases for it. 

To give a high level overview, the key functionality is: 

- Issues can exist within multiple boards so that different teams (or different projects within the same team) can operate over distinct workflows. Where workflow relates the the current 'progress and priority' of an issue.  

- This means that an issue can exist 'in two places at once'. For example, it may be 'in progress' for the development team, but in the 'backlog' for a QA team. 

- Not all issues from a repo would want to appear in the new board, so a team should be able to set a filter for just the issues they are interested in. 


**Step 1:** Update the ZenHub UI to give us room to create multiple boards per repo:

This will be done by moving to a sidebar under a single 'ZenHub' tab, rather than the existing 'Boards' and 'Reports' tabs:

Expanded:
![image](https://user-images.githubusercontent.com/7584089/34697215-fc9d75a4-f486-11e7-8dbe-2922707d9668.png)


Collapsed:
![image](https://user-images.githubusercontent.com/7584089/34697239-1523374e-f487-11e7-90cf-918cabcc2566.png)

**Step 2:** Introduce the concept of the 'Workspace' which can be created by one or more repositories. This replaces the current terminology of 'multi-repo-board' and allows a repo to be added to as many workspaces as a team needs. A workspace can be named and customized as required. 

The sidebar introduces a new selector to show which workspace you are operating in, here we can see that the ZenHub Design repo is part of two workspaces 'Design' and 'ZenHub':

![image](https://user-images.githubusercontent.com/7584089/34697652-eb9c717c-f488-11e7-96fa-2a094be3e378.png)

All the sidebar content, such as 'boards' and 'reports' will then be applicable to the selected workspace. 

When building a workspace, it should be possible to select to customize the workspace:

![image](https://user-images.githubusercontent.com/7584089/34697771-7d778e88-f489-11e7-9791-7fe05d809917.png)

For example, by only showing issue with a certain label attached to them:

![image](https://user-images.githubusercontent.com/7584089/34697791-9597c2ee-f489-11e7-97a1-e23599991af4.png)


Once live, I think this then opens up a lot of interesting automation ideas for issues to be passed between teams. But lets get the core functionality live first 😄 


Please let me know your thoughts on this, would it resolve the root problems of this issue for you and your team, or are we missing something? 

Also happy to jump on calls for more detailed feedback, please just email support@ZenHub.com and we can find a time.

**Comment by MilesRomney** [comment]
Yes! This looks great.

> Once live, I think this then opens up a lot of interesting automation ideas for issues to be passed between teams. But lets get the core functionality live first

Absolutely. Even something as simple as adding/removing a tag when an issue moves in/out of a pipe within a workspace would be great, when paired with the filter you describe. For example, when an item moves into the "Staged" pipe in the Development workspace, automation would add a tag that makes it visible in the "Ready to Test" pipe in the QA workspace.

Good stuff.

**Comment by erichexter** [comment]
This looks very promising and you may be one of the few products to
implement this capability.

**Comment by mcphersonwhite** [comment]
This feature would be huge, and with the integration into GitHub UI, would remove our need of Trello.

**Comment by loewenstein-zz** [comment]
Happy to see this going forward in favor of closing #950. I think you got it even without further explaining it there. Thank you. I am keen on seeing this deployed in our GHE instance somewhen.

**Comment by tomgatzgates** [comment]
This is a fantastic feature especially for larger organizations with shared repos, I'm really looking forward to seeing it roll out. @azenMatt is there a beta that could be opted into?

**Comment by tio-max** [comment]
I am going to ditch Jira Cloud when this feature goes live. What is the planning for delivery of this feature? Before or after summer?

**Comment by azenMatt** [comment]
@tomgatzgates - can you drop an email to support@zenhub.com and we would be happy to let you know if we release part of this as a beta. Ideally we will be aiming to release in cycles to production over the next couple of months. But more feedback is always better! 

@tio-max - no fixed release estimate at this point, but the work is in progress so we should have some firmer updates soon

**Comment by laurencesmithever** [comment]
This. Will. Change. Everything.

**Comment by Charlenes1** [comment]
Watching - really need this!

**Comment by lucasdidier** [comment]
+1

**Comment by ibm-jeff** [comment]
@azenMatt I'm pretty new to github and zenhub, but was trying to figure out how to create a new Board for our project (we already have one board, but I wanted to track a different aspect of our project)... apparently not supported yet, but per this issue, hopefully soon...   My newbie comments on your functionality statement:

To give a high level overview, the key functionality is:

Issues can exist within multiple boards so that different teams (or different projects within the same team) can operate over distinct workflows. Where workflow relates the the current 'progress and priority' of an issue.

This means that an issue can exist 'in two places at once'. For example, it may be 'in progress' for the development team, but in the 'backlog' for a QA team. **I would think of this more like 'in progress' with regard to one board, but in the 'backlog' with regard to another board.**

**Also, hopefully each board created can specify its own set of repos included.**

Not all issues from a repo would want to appear in the new board, so a team should be able to set a filter for just the issues they are interested in.  **Yes, for example in my case, I am using a label on all issues that I want to see in this new board I'd like to create.**

**I'll give you a good use-case to consider... say you have a typical SaaS product undergoing normal set of iterative development... Now you find you need to comply with a new standard, like ISO 12345... so you create a bunch of issues to implement that compliance, and the Project Mgr is tasked with tracking progress on compliance across multiple repos and multiple iterations.... the project already has a board in place, and now this PM wants to create a new ISO 12345 status board that spans repos and iterations. 
 :-) ....Is it ready yet??? **

[Anything else core that we we have missed?]

**Comment by andrewkelly83** [comment]
@azenMatt Is there a timeline on the delivery of this feature. It is something we are crying out for.

**Comment by Dyslexion** [comment]
really need this!! we have multiple teams on the same product and having boards that can be attached to the same repo would be AMAZING

**Comment by Dyslexion** [comment]
is this actually getting worked on?

**Comment by DanielePucci** [comment]
Hi @azenMatt ! Sorry for bothering, just wanted to know if there was any update on this! Relying on your [comment above](https://github.com/ZenHubIO/support/issues/139#issuecomment-366020590), the party for this new feature should begin soon 😄 !

**Comment by azenMatt** [comment]
> is this actually getting worked on?

It is in progress right now! 

@ibm-jeff - spot on!

**Comment by burtonator** [comment]
This is kind of important to use too.. it's a main feature of trello.  Comparing you vs trello there are some features I like there and multiple boards is one of them.

**Comment by azenMatt** [comment]
Hi everyone, 

We are still working on this change and are looking to have a beta version that everyone here can test in a few months. 

In the mean time, we have a potentially releasable slice that mainly updates the navigation and UI as building blocks towards this release. We are currently discussing if this is something that provides enough value to release by itself, or it we should wait until the full beta is available. 

This is where everyone here comes in! Please let me know if you have any feedback on the changes below and if you would find them valuable to be released as-is or if you would rather stay with the existing version until more functionality is present. I'm also happy to demo this change to anyone who would like to jump onto a call to discuss - please email me - support@zenhub.com

The main changes this release would contain are: 

1) Starting to move the concept of a Multi-repo-board (which is what we call the board type where you can add repos together, but each repo can only belong to one) towards the concept we are calling 'workspaces' (eventually repos will be able to exist in multiple workspaces. For this release it is still the same as a multi-repo-board but with new naming)

As part of this move, you will now be able to name your workspace and add a description. 

For example, here we can see a workspace consisting of the two repos 'communications' and 'marketing-events'. It is currently untitled, which means if you want to, you can provide a unique name for it. The idea here is to be able to more easily direct team-members to different parts of ZenHub. 

![image](https://user-images.githubusercontent.com/7584089/41567849-540cbfd4-7317-11e8-8874-01222617581d.png)

On the edit screen for this, a name and description can be added:

![image](https://user-images.githubusercontent.com/7584089/41567867-72e66fd6-7317-11e8-88c9-b6d40155a707.png)

Which is then reflected on the side bar:

![image](https://user-images.githubusercontent.com/7584089/41568082-8e997fa6-7318-11e8-874d-c296c9359345.png)


The repo switcher component of the side bar itself, (entered when searching for a workspace), will be updated to look like:

![image](https://user-images.githubusercontent.com/7584089/41567901-a4fdb63c-7317-11e8-8e56-523a05f4ffe8.png)

Where you can search for a workspace or repo, as well as see favourites and recently viewed workspaces/repos. We hope the addition of descriptions here will help locate the right places for the right things.


Finally, on the webapp only, the top bar will be updated to show you the workspaces name when the sidebar is collapsed.

![image](https://user-images.githubusercontent.com/7584089/41567042-2d7f3d96-7313-11e8-8a75-e75e2cb56e3c.png)


------

Let me know your thoughts, 

Either way, we are still full steam ahead on getting to the beta version of this change and will keep you all updated with progress on that!

**Comment by tio-max** [comment]
want!

**Comment by andrewkelly83** [comment]
@azenMatt it’s looking good, we really need this to help us out when working with external teams. We’d be able to put all their deliverables in its own workspace but still see it all move along as part of a sprint

**Comment by MilesRomney** [comment]
Anxious for the finished feature.

I don't think that simply renaming Multi-repo Boards to Workspaces is a useful WIP release, though. Until multiple Workspaces are supported, and a repo can live in multiple Workspaces, it won't be useful for my team.

God speed.

**Comment by nunoguedelha** [comment]
If having two issues in the same workspace is enough for defining a dependency between them, or even to add both to a common Epic, then this update is **very useful for our team**.

**Comment by azenMatt** [comment]
> If having two repos in the same workspace is enough for defining dependencies between their respective issues, or even to create Epics with issues from both repos, then this update is very useful for our team.

Hi @nunoguedelha - you should be able to do this already today via https://www.zenhub.com/guides/creating-a-multi-repo-board please feel free to reach out to support@zenhub.com if you need any help with that.

**Comment by ryandeussing** [comment]
This all looks good and I appreciate the status update but I'm also of the opinion that until we can have separate workspaces/boards referencing the same repo, there isn't any functionality to release here.

**Comment by jasell** [comment]
Workspace is not an ideal name. Workspace is in my experience something that is private (one user only).
This is based on several other systems ranging over several disciplinens from the 90’s until now.

**Comment by bmunkholm** [comment]
If I understand this correctly, to boil it down the main improvement this would be is in the area of board-selection and to be able to see/select recently used boards and have favorites?
That would be useful, so I don't know why not to release that. Unless you feel the whole concept will be too confusing for new users. But I would be fine with it.

**Comment by nunoguedelha** [comment]
> Hi @nunoguedelha - you should be able to do this already today via https://www.zenhub.com/guides/creating-a-multi-repo-board please feel free to reach out to support@zenhub.com if you need any help with that.

@azenMatt sorry for the confusion, I meant: having initially two repos belonging to separate boards, and still be able to include them in a common workspace and then be able to connect the issues from those repos.
My main concern is that we cannot connect in any way issues from repos belonging to different boards.

**Comment by azenMatt** [comment]
Thanks all for the feedback here. Based on this it seems that the most useful thing for the ZenHub team to do is focus on getting to the full workspaces release. So we are going to avoid any intermediate changes like the one I described above until we can put something live that tackles the core one repo in two places problem

**Comment by ryandeussing** [comment]
I made the mistake of mentioning to colleagues almost a year ago that this feature was coming. Now those same people are about to drag us back to Jira.

**Comment by andrewkelly83** [comment]
Is there a atimeline for delivery on this? It feels like it has been sat in progress for a very long time

**Comment by kendramoroz** [comment]
Hi everyone - I appreciate that it's been frustrating without updates from our end. The team is still in active development of this functionality at this time. We'll be discussing more concrete timelines to share next week.

**Comment by DanielePucci** [comment]
@kendramoroz 
Any news on the timeline?

**Comment by tio-max** [comment]
@kendramoroz  you promised us a timeline a month ago. Any word?

**Comment by kendramoroz** [comment]
🤕 Thanks for holding us accountable, @tio-max - We have invested significant amount of planning time in the last few weeks to unpack timelines and get a shared understanding of all the upcoming architectural changes upcoming. We are aiming to have a full timeline of upcoming phases of launching workspaces, what each phase will include, and a shared timeline of when to expect the beta. We're aiming to deliver this by early next week. 

We know it's been a dark hole and longer than expected timelines, and when paired together these do not make for good communication. We acknowledge this as a team, and know this is not okay, so please know we're working to not just provide this visibility for workspaces and this feature, but are aggressively improving our feedback process, communication strategy for our roadmap, and have a keen eye on improvement.

We appreciate the patience y'all have shared with us, and know that we are excited about introducing stronger communication and more visibility into where we're going!

**Comment by jasell** [comment]
Thanks for breaking the silence, were are looking forward to this communication.
We are not only waiting for this feature eagerly, but we are also curious to see the improved communication. How can we all learn from this.

I would like emphasis on my early note to find a better, more suitable, name for this feature. Workspace are recognized as something private, something for one user only.

Having multiple boards for one repo is more of a phasing, roadmap or organizational feature,  and the name should reflect that.

**Comment by MilesRomney** [comment]
You could call it Purple Unicorns for all I care, so long as I get the feature. :)

> Workspace are recognized as something private, something for one user only.

Hm. That has never been my impression. Many products brand collaborative spaces as Workspaces: WeWork, Oblong, DisplayNote...

"Workspace" is fine with me.

**Comment by kendramoroz** [comment]
👋 Hey all - As promised, an update!

[Access here a detailed plan of our launch plan](https://view.attach.io/H1TPiia-7) for both Workspaces, and the upcoming Kanban reporting work we are mid-development on.

The team would like to also take the opportunity to share communication commitments:
- We will be introducing an improved feedback request / feedback loop process and forum. This will prevent stagnancy of requests and help streamline communications with regards to what's under consideration, in progress, or available for beta testing.
- A product communication commitment introduces detailed project plans (from the beginning of a project moving forward!), at a minimum quarterly product roadmaps, and improved communication on beta program timelines. 

Check out [the roadmap doc](https://view.attach.io/H1TPiia-7) and let us know if you have any questions. We hope this helps bring much-needed visibility into the phases of workspaces and timelines to expect updates!

**Comment by MilesRomney** [comment]
Good stuff in here. Thanks for sharing!

**Comment by osmanis** [comment]
Any update on this issue? According to previous comments it's been in development since 2014.

**Comment by kendramoroz** [comment]
👋 It's been a great journey here in this GitHub Issue! We're going to be sunsetting this repo and moving to a new product feedback portal, [hosted here](https://portal.productboard.com/zenhub/c/52-have-repos-live-in-multiple-repos-have-multiple-workflows-on-the-same-repo?utm_medium=social&utm_source=portal_share) to streamline requests and share our roadmap more openly.

We encourage everyone interested in this next chapter of Workspaces and ZenHub to vote, add feedback, and join the conversation 🎄! Upwards and onwards to 2019 🚀
